File: Source/DotNET/Model/Interactions/RefreshQueryAction.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Cratis.Scene.Model.Interactions;

/// <summary>
/// Re-runs a query-backed element.
/// </summary>
/// <param name="Query">The query to re-run.</param>
public record RefreshQueryAction(string Query) : InteractionAction;

File: Source/JavaScript/model/interactions/RefreshQueryAction.ts
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

import { InteractionAction } from './InteractionAction';

/**
 * Re-runs a query-backed element.
 */
export interface RefreshQueryAction extends InteractionAction {
    query: string;
}

export const RefreshQueryActionPropertyNames: (keyof RefreshQueryAction)[] = ['query', 'kind'];
